content/00.build/40.tooling/20.hardhat/20.migrating-to-zksync.md

```diff
@@ -1,275 +1,527 @@
 
-import "@matterlabs/hardhat-zksync";
+::callout{icon="i-heroicons-information-circle" color="blue"}
+Make sure the `@matterlabs/hardhat-zksync` plugin is installed and the networks configured as detailed in the [project setup](#project-setup).
+::
 
-const config: HardhatUserConfig = {
-  zksolc: {
-    version: "latest", // Uses latest available in %%zk_git_repo_zksolc-bin%%
-    settings: {},
-  },
-  defaultNetwork: "zkSyncTestnet",
-  networks: {
-    hardhat: {
-      zksync: false,
-    },
-    sepolia: {
-      url: "https://sepolia.com/api/abcdef12345",
-      zksync: false,
-    },
-    mainnet: {
-      url: "https://ethereum.mainnet.com/api/abcdef12345",
-      zksync: false,
-    },
-    zkSyncTestnet: {
-      url: "%%zk_testnet_rpc_url%%",
-      ethNetwork: "%%zk_testnet_identifier%%", // or a Sepolia RPC endpoint from Infura/Alchemy/Chainstack etc.
-      zksync: true,
-    },
-  },
-  solidity: {
-    version: "0.8.13",
-  },
-  // OTHER SETTINGS...
-};
+### Deployment with `hardhat-ethers`
+
+The `@matterlabs/hardhat-zksync` includes `@matterlabs/hardhat-zksync-ethers`, a package that extends `@nomiclabs/hardhat-ethers`
+with all the necessary helper methods to deploy contracts on both ZKsync and EVM networks. The injected `hre.ethers`
+object provides methods like `deployContract`, `getContractFactory` or `getContractAt` so deployment scripts work out of the box.
 
-export default config;
+::callout{icon="i-heroicons-exclamation-triangle" color="amber"}
+To avoid typescript collision errors between `@nomiclabs/hardhat-ethers` and `@matterlabs/hardhat-zksync-ethers` make sure that only the latter
+(or the wrapper plugin `@matterlabs/hardhat-zksync`) is imported in the `hardhat.config.ts` file.
+::
+
+See below examples for `hardhat-ethers` and `hardhat-zksync-ethers`:
+
+::code-group
+
+```typescript [hardhat-ethers]
+const greeter = await hre.ethers.deployContract('Greeter', ['Hi there!']);
+await greeter.waitForDeployment();
 ```
 
-## Compile contracts
+```typescript [hardhat-zksync-ethers]
+const greeter = await hre.ethers.deployContract('Greeter', ['Hi there!']);
+await greeter.waitForDeployment();
+```
 
-To compile your contracts for ZKsync Era, run:
+::
+
+In-case a custom deployment is needed, use `ContractFactory`.
 
 ::code-group
 
-```bash [yarn]
-yarn hardhat compile --network zkSyncTestnet
+```typescript [hardhat-ethers]
+const GreeterFactory = await hre.ethers.getContractFactory('Greeter');
+const greeterContract = GreeterFactory.deploy(); // if any, pass constructor arguments in deploy arguments
+await greeter.waitForDeployment();
 ```
 
-```bash [npm]
-npm hardhat compile  --network zkSyncTestnet
+```typescript [hardhat-zksync-ethers]
+const GreeterFactory = await hre.ethers.getContractFactory('Greeter');
+const greeterContract = GreeterFactory.deploy(); // if any, pass constructor arguments in deploy arguments
+await greeter.waitForDeployment();
```

Comment:
All of these code examples have the same code between the hardhat-ethers and hardhat-zksync-ethers. Is that expected?

Reply:
yes. I want to highlight that it's the same in EVM and ZKsync